test/Libraries/TestServices/TestSessionConfiguration.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Configuration;
+using System.IO;
+using System.Reflection;
+
+using DynamoShapeManager;
+
+namespace TestServices
+{
+    /// <summary>
+    /// The TestSessionConfiguration class is responsible for 
+    /// reading the TestServices configuration file, if provided,
+    /// to provide data for configuring a test session from a location
+    /// other than Dynamo's core directory.
+    /// </summary>
+    public class TestSessionConfiguration
+    {
+        private const string CONFIG_FILE_NAME = "TestServices.dll.config";
+
+        public string DynamoCorePath { get; set; }
+        public LibraryVersion RequestedLibraryVersion { get; set; }
```

Comment:
These are no longer `private set`, their values are assigned during construction, any caller that requires these to be set should be redesigned, IMO. If we allow those callers to alter these properties, then the process is no longer centralized, those callers will have their own logic and once `TestSessionConfiguration` needs updating, those other parts better be updated.